src/main/java/org/flyak/api/controllers/AuthController.java

```diff
@@ -36,15 +46,47 @@
     @Value("${app.registration.enabled}")
     private Boolean registrationEnabled;
     private Logger log = LoggerFactory.getLogger(AuthController.class);
+    @Value("${app.ui.baseurl}")
+    private String UIBaseURL;
+    @Value("${app.ui.registration_verification}")
+    private String UIRegVerificationURL;
+    private TokenGenerator tokenGenerator;
+    private EmailService emailService;
+    private UserRepository userRepository;
 
-    public AuthController(AuthService authService, AuthenticationManager authenticationManager, JwtUtils jwtUtils) {
+    public AuthController(AuthService authService,
+                          UserRepository userRepository,
+                          AuthenticationManager authenticationManager,
+                          JwtUtils jwtUtils,
+                          EmailService emailService) {
         this.authService = authService;
         this.authenticationManager = authenticationManager;
+        this.userRepository = userRepository;
         this.jwtUtils = jwtUtils;
+        this.tokenGenerator = new TokenGenerator(12, new SecureRandom());
```

Comment:
Why 12? And Should we move these magic numbers / variables into constants or config type variables?